src/csharp/Meadow/DebugSessionLogger.cs

```diff
@@ -24,20 +27,34 @@ public bool IsEnabled(LogLevel logLevel)
 			if (System.Diagnostics.Debugger.IsAttached)
 				return true;
 
-			return logLevel >= LogLevel.Information;
+			return logLevel >= _minLogLevel;
 		}
 
 		public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
 		{
-			if (IsEnabled(logLevel))
+			if (!IsEnabled(logLevel)) {
+				return;
+			}
+
+			if(formatter == null) {
+				throw new ArgumentNullException(nameof(formatter));
+			}
+
+			try
+			{
+				var message = formatter(state, exception);
+				Callback?.Invoke(message);
+			}
+			catch (Exception ex)
 			{
-				Callback?.Invoke(formatter(state, exception));
+				Console.WriteLine($"Logging failed: {ex.Message}");
+				throw;
 			}
 		}
 
 		internal async Task ReportDeviceMessage(string source, string message)
 		{
-			this.LogInformation($"{source}: {message}");
+			await Task.Run(() => this.LogInformation($"{source}: {message}"));
```

Comment:
Why run this in an awaited task?  If LogInformation is a synchronous call, then it's going to block either way?

Reply:
@ctacke  You're absolutely right. No idea what I was thinking. Brain fart! I've made logging all sync in latest push.